x/interchainstaking/keeper/grpc_query.go

```diff
@@ -254,3 +258,33 @@ func (k *Keeper) RedelegationRecords(c context.Context, req *types.QueryRedelega
 
 	return &types.QueryRedelegationRecordsResponse{Redelegations: redelegations}, nil
 }
+
+func (k *Keeper) MappedAccounts(c context.Context, req *types.QueryMappedAccountsRequest) (*types.QueryMappedAccountsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	var mappedAccounts []types.MappedAccount
+
+	k.IterateZones(ctx, func(index int64, zone *types.Zone) (stop bool) {
```

Comment:
Would it not be far more efficient to just use a prefixed iterator over the mappedAccounts and return a map of key/account?

Reply:
e.g. we index the mapped addresses by:

prefix + local address bytes + chain  == > remote address bytes

if we iterate over:

prefix+local_address_bytes

we can return a map of chain_id (from key): remote_address_bytes

we don't have to filter at all here, because the every matching KV is correct for that user.

Reply:
where do we get chain_id's from? the query request only has local address -> do i change that as well?

Reply:
[ddc2dc9](https://github.com/ingenuity-build/quicksilver/pull/443/commits/ddc2dc99988b7f41ae5fc5640befe1be2124f283)